rules/do-you-use-logging-fakes/rule.md

```diff
@@ -0,0 +1,142 @@
+---
+seoDescription: Learn how to use Microsoft's `FakeLogger` in unit tests to verify logging behavior in .NET applications. This guide shows you how to capture log entries, check log levels, and ensure your logging works as expected during testing.
+type: rule
+title: Do you use Logging Fakes?
+uri: do-you-use-logging-fakes
+authors:
+  - title: Tom Iwainski
+    url: https://www.ssw.com.au/people/thomas-iwainski
+related: []
+created: 2024-11-05T12:52:08.000Z
+archivedreason: null
+guid: dc76970f-deea-4e92-9d55-731e83bee7b5
+---
+
+Unit tests are an essential part of ensuring that your .NET applications work correctly. However, when it comes to testing logging, things can get tricky. You don't want to rely on actual logging infrastructure like file writes or external logging systems during tests, as this could lead to unnecessary complexity and slow down your test suite.
```

Comment:
```suggestion
Unit tests are an essential part of ensuring .NET applications work correctly. However, when it comes to testing logging, things can get tricky. You don't want to rely on actual logging infrastructure like file writes or external logging systems during tests, as this could lead to unnecessary complexity and slow down your test suite.
```